enhancements/workload-partitioning/pao-render.md

```diff
@@ -0,0 +1,190 @@
+1) When a command line option `--enable-workload-partitioning` is passed to PAO in render mode, it also generates the necessary configuration for the `management` partition to work properly
+1) PAO at day 2 must not make any changes to the existing generated objects it owns
+1) PAO sets the ownership references on the generated yamls so the generated files are properly linked to the PerformanceProfiles
+
+### Non-Goals
+
+TBD
+
+## Proposal
+
+### User Stories
+
+- 1) [optional] The cluster administrator will collect PAO must-gather output from a test cluster with no partitioning enabled
+  1) [optional] The cluster administrator will execute the Performance profile creator to get well formed PerformanceProfiles
+  1) The PerformanceProfiles are placed into a directory
+  1) PAO is executed using podman run -v /input/dir:/input -v /output/dir:/output registry/performance-addon-operator render [optional --enable-workload-partitioning]
+  1) The files from both input and output directories are passed over to the installer
+
+### Implementation Details/Notes/Constraints [optional]
+
+TBD
+
+### Risks and Mitigations
+
+- PAO needs to be written in such way that the render mode and the live reconcile loop always generate identical content. This should be handled by tests and/or CI.
+
+## Design Details
+
+### Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance,
+ > 1. This requires exposing previously private resources which contain sensitive
+  information.  Can we do this?
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+- What additional testing is necessary to support managed OpenShift service-based offerings?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+This feature is supposed to be GA and supported right from release. It requires an explicit user action to be activated.
+
+The plan is to provide tests, CI and documentation in the release where this will be introduced.
+
+End-to-end tests of the Workload partitioning are covered in the high level feature description.
+
+### Upgrade / Downgrade Strategy
+
+This feature is activated locally by user and follows the lifecycle of PAO itself. There is no upgrade procedure to perform.
+
+### Version Skew Strategy
```

Comment:
Do we need to worry about matching the PAO version and OCP version? I think we talked about that, but don't remember whether we came to a conclusion.

Reply:
Maybe the desired version could be passed into the render command so there could be validation against the running image.

Reply:
Well, the desired version is part of the PAO pull spec, so the validation would basically compare whether a command line argument matches the compiled in version string. We do not support generating older versions, you need to use older PAO for that.

Reply:
We should list the skew as a risk. We're going to want to make sure the documentation at least explains that the versions have to match.